Reconstruction of ServiceControl menu (#5437)
Particular/docs.particular.net: servicecontrol/installation.md

 - servicecontrol/multi-transport-support
---

Every component in the Particular Service Platform, including ServiceControl, needs to be [downloaded](https://particular.net/downloads) and installed.

> After installation there is no actual ServiceControl instance running yet. Those instances can be installed, upgraded and removed with the ServiceControl Management Utility. This utility is launched as the final step in the installation process and is also available via the Windows Start Menu.

Reviewer: ```suggestion
After installation, there is no ServiceControl instance running yet. These instances can be installed, upgraded, and removed by making use of the ServiceControl Management Utility. This utility is launched as the final step in the installation process and is also available via the Windows Start Menu.
```

* [Microsoft .NET 4.7.2 Runtime](https://dotnet.microsoft.com/download/dotnet-framework/net472)

## Planning

> WARNING: In production be aware of the possible considerations when setting up a machine with ServiceControl.

Reviewer: Shouldn't we be linking to those considerations here?

Reviewer: ```suggestion
WARNING: In production environments, make sure to review the [environment considerations](/servicecontrol/servicecontrol-instances/hardware.md) when setting up a machine with ServiceControl.
```

> ServiceControl Management provides a simple means of setting up one or more instances of ServiceControl service types (error, audit, and monitoring). For production systems, it is recommended to limit the number of instances per machine to one. The ability to add multiple instances *of the same type on a single machine* is primarily intended to assist development and test environments.

Reviewer: ```suggestion
ServiceControl Management provides a simple way to set up one or more ServiceControl instances (error, audit, and monitoring). For production systems, it is recommended to limit the number of instances per machine to one of each type. The ability to add multiple instances *of the same type on a single machine* is primarily intended to assist development and test environments.
```

See [ServiceControl Capacity Planning](capacity-and-planning.md) and [Hardware Considerations](/servicecontrol/servicecontrol-instances/hardware.md) for more guidance.

## Installing ServiceControl instances

There are [three types](/servicecontrol/#servicecontrol-instance-types) of ServiceControl instances that can be installed using the ServiceControl Management utility. As the error and audit instance usually go side-by-side, they are installed and configured at the same time.

> 1. Open the ServiceControl Management Utility in the Windows start menu.
> 1. Click the `New` button at the top-right and a popup window appears.
> 1. Select either `Add ServiceControl and Audit instances` or `Add monitoring instance`.
> 1. Provide a name and transport.
>    1. The default name commonly used is `Particular.ServiceControl`.  
>       It has no real use except for the fact that it is being used to finalize the name for the error and audit instances. The name of each instance can be adjusted from its default if required.
>    1. The transport should be the same as the transport the endpoints are using.  
>       If a transport is not available, a [transport adapter](/servicecontrol/transport-adapter/incompatible-features.md) can be used.
> 1. For additional instance settings, open the `ServiceControl` and `ServiceControl Audit` sections.
>    1. Here the name of the instance can be altered, the defaults for error and audit are `Particular.ServiceControl` and `Particular.ServiceControl.Audit`.
>       1. The name for the error instance is particularly important to [enable plugins to send information](/servicecontrol/installation.md#servicecontrol-plugins) to ServiceControl.
>       1. If multiple instances for different systems are installed on the same server, a name like `Particular.SystemName` could be an option as well.
>    1. Select the appropriate user account.  
>       Be aware that ServiceControl instances will run as Windows Services in the background.
>    1. Be aware of the port numbers as these are used by ServicePulse and ServiceInsight to connect to ServiceControl.
>    1. Configure the [retention period](/servicecontrol/how-purge-expired-data.md) for each instance.
>    1. Configure the name of the queue that messages should arrive in.  
>       This queue is important to endpoints that should send error and audit messages to these ServiceControl instances, or for [plugins](/servicecontrol/installation.md#servicecontrol-plugins).
>    1. Potentially configure [forwarding queues](/servicecontrol/errorlog-auditlog-behavior.md).
>    1. Full-text search can be turned off for [performance reasons](/servicecontrol/capacity-and-planning.md#storage-performance).

Reviewer: ```suggestion
1. Open the ServiceControl Management Utility in the Windows start menu.
1. Click the `New` button at the top-right and a popup window appears.
1. Select either `Add ServiceControl and Audit instances` or `Add monitoring instance`.
1. Provide a name and transport.
   1. The default name commonly used is `Particular.ServiceControl`.  
      The name is used to derive names for the error and audit instances. The name of each instance can be adjusted from its default if required.
   1. The transport should match the endpoint's transport.
      If a transport is not available, a [transport adapter](/servicecontrol/transport-adapter/incompatible-features.md) can be used.
1. For additional instance settings, open the `ServiceControl` and `ServiceControl Audit` sections.
   1. The name of the instance can be altered, the defaults for error and audit are `Particular.ServiceControl` and `Particular.ServiceControl.Audit`.
      1. The name for the error instance is particularly important to [enable plugins to send information](/servicecontrol/installation.md#servicecontrol-plugins) to ServiceControl.
      1. If multiple instances for different systems are installed on the same server, a name like `Particular.SystemName` could be an option as well.
   1. Select the appropriate user account.  
      Be aware that ServiceControl instances will run as Windows Services in the background.
   1. Be aware of the port numbers as these are used by ServicePulse and ServiceInsight to connect to ServiceControl.
   1. Configure the [retention period](/servicecontrol/how-purge-expired-data.md) for each instance.
   1. Configure the name of the queue that messages should arrive in.  
      This queue is important to endpoints that send error and audit messages to these ServiceControl instances, as well as [plugins](/servicecontrol/installation.md#servicecontrol-plugins).
   1. If needed, configure [forwarding queues](/servicecontrol/errorlog-auditlog-behavior.md).
   1. Full-text search can be turned off for [performance reasons](/servicecontrol/capacity-and-planning.md#storage-performance).
```

Reviewer: What is meant by:

> If a transport is not available, a [transport adapter](/servicecontrol/transport-adapter/incompatible-features.md) can be used.

A monitoring instance differs from error and audit instances in its configuration:

> 1. Configuration
>    1. There is no configuration for storage as it only stores data in-memory.
>    1. There is no forwarding queue as the messages are specific to ServiceControl monitoring.
>    1. The queue to forward messages to that can't be processed, as described in [monitoring setup](/tutorials/monitoring-setup/#component-overview).

Reviewer: ```suggestion
1. There is no configuration for storage as it only stores data in memory.
1. There is no forwarding queue as the messages are specific to ServiceControl monitoring.
1. The queue to forward messages to that can't be processed, as described in [monitoring setup](/tutorials/monitoring-setup/#component-overview).
```

After configuring an error, audit and monitoring instance for the SQL Server transport, the ServiceControl Management Utility should look similar to this:

![ServiceControl Management interface that shows the link for upgrading](managementutil.png 'width=500')

## Upgrading ServiceControl instances

Whether a new version of ServiceControl is available can be verified in the ServicePulse user interface. A new version can then be [downloaded](https://particular.net/downloads) and installed.

> ServiceControl Management will display the instances of the ServiceControl service installed. If the version used by an instance are older than those that were downloaded and installed, an upgrade link will be shown next to the version label.

Reviewer: ```suggestion
ServiceControl Management will display the instances of the ServiceControl service installed. If any ServiceControl instance is running on an outdated version, an upgrade link will be shown next to the version label.
```

* Run the new binaries to create required queues
* Start the service

## ServiceControl plugins

> Endpoint plugins like heartbeat and custom checks require sending information to ServiceControl. The name of the queue is the exact name of the error instance.

Reviewer: ```suggestion
Endpoint plugins like heartbeats and custom checks require sending information to ServiceControl. The name of the queue is the exact name of the error instance.
```
